cdap-proto/src/main/java/co/cask/cdap/proto/artifact/ArtifactVersion.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.artifact;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Represents version of an artifact. It will try to parse the version string of format
+ *
+ * <pre>
+ * {@code
+ * parse version string in format of [major].[minor].[fix](-|.)[suffix]
+ * }
+ * </pre>
+ */
+@Beta
+public final class ArtifactVersion implements Comparable<ArtifactVersion> {
```

Comment:
This class looks similar to the PluginVersion class. Will we remove the latter later?